workspace/document.go

```diff
@@ -0,0 +1,34 @@
+package workspace
+
+import "path/filepath"
+
+// Document is a file in a directory.
+type Document struct {
+	Root         string
+	RelativePath string
+}
+
+// NewDocument creates a document from a relative filepath.
+// The root is typically the root of the exercise, and
+// path is the relative path to the file within the root directory.
```

Comment:
Path is no longer a relative path.  Path needs to be an absolute path, rooted with root, or filepath.Rel will return an error. 

calling filepath.Abs before calling filepath.Rel in this method is an option but it can result in unwanted behavior. https://play.golang.org/p/cLE1R9dsRcH


I think a `filepath.HasPrefix(root, path)`  check before .Rel would catch it.